linopy/constraints.py

```diff
@@ -872,6 +879,27 @@ def sanitize_missings(self):
                 contains_non_missing, -1
             )
 
+    def sanitize_infinities(self) -> None:
+        """
+        Replace infinite values in the constraints with a large value.
+        """
+        for name in self:
+            constraint = self[name]
+            invalid_infinity_values = (
+                (constraint.sign == LESS_EQUAL) & (constraint.rhs == -np.inf)
+            ) | ((constraint.sign == GREATER_EQUAL) & (constraint.rhs == np.inf))
+            if invalid_infinity_values.any():
+                raise ValueError(
+                    f"Constraint {name} contains incorrect infinite values."
+                )
```

Comment:
Hmm, this check is kind of not related to sanitizing and might make more sense during Constraint initialization? So it can not turned off? Might also be not nice to split that up

Reply:
you are right, moved it to the constraint assignment